### PR TITLE
Fix dune stanza for stdlib: change of name

### PR DESCRIPTION
**Pull request by HyunggyuJang**
With the commit of 44c9929fa988d3ba43ce5ffe6e124c35da3580b8, the file `camlinternalAtomic.ml` is removed. Hence, dune file also has to be updated properly.

I added necessary amend to the `stdlib/dune`.

### CODE REVIEW

File: stdlib/dune [hunk 1]
```diff
@@ -22,7 +22,11 @@
  (flags (:standard -w -9 -nolabels))
  (preprocess
    (per_module
-     ((action (run awk -f %{dep:expand_module_aliases.awk} %{input-file}))
+    ((action
+      (progn
+       (run sed -i s/%atomic_load/%identity/ %{input-file})
+       (run sed -i s/%atomic_cas/%obj_set_field/ %{input-file})
+       (run awk -f %{dep:expand_module_aliases.awk} %{input-file})))
```
Comment (dra27) [inline stdlib/dune:29]: There's definitely no need to call `sed` twice, but I'm also not sure that Dune expects this file to being altered in this way (i.e. having a pre-processor modify the input file - it's meant to pre-process it to stdout).  I would expect that changing the `lang dune` in `dune-project` to 2.7 and then:

```suggestion
    ((action
      (pipe-stdout
        (run sed s/%atomic_load/%identity/;s/%atomic_cas/%obj_set_field/ %{input-file})
        (run awk -f %{dep:expand_module_aliases.awk})))
```
should work better.

All that said, I'm not clear why this change is necessary at all!

Comment (HyunggyuJang) [reply stdlib/dune:29]: Thanks for suggestion. As you said, using `pipe-stdout` can be adopted only after changing `dune-project` to 2.7.
For the reason why this change is needed, I believe that as of #11038 the primitives needing awful hacks are introduced in `stdlib.ml` also.

**Review by dra27 — APPROVED** [review]
Ah, apologies - I had reviewed the diff and not taken in the context properly that these hacks are already done in other files. It's temporary, and this is for developing the compiler, so there's no need to improve it further (at least now!)
